internal/storage/v2/elasticsearch/tracestore/reader.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package tracestore
+
+import (
+	"context"
+	"iter"
+
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/internal/storage/elasticsearch/dbmodel"
+	"github.com/jaegertracing/jaeger/internal/storage/v1/elasticsearch/spanstore"
+	v2api "github.com/jaegertracing/jaeger/internal/storage/v2/api/tracestore"
+)
+
+type TraceReader struct {
+	spanReader spanstore.CoreSpanReader
+}
+
+func NewTraceReader(p spanstore.SpanReaderParams) *TraceReader {
+	return &TraceReader{
+		spanReader: spanstore.NewSpanReader(p),
+	}
+}
+
+func (t *TraceReader) GetTraces(ctx context.Context, traceIDs ...v2api.GetTraceParams) iter.Seq2[[]ptrace.Traces, error] {
```

Comment:
you don't need to implement every single function of the API in a single PR, it will make the tests very large. Start with services / operations, which are relatively trivial, and return not-implemented from the other functions.

This is alway the way not to get blocked on a single review, you can work on those independently.